lib/oidcstrategy.js

```diff
@@ -866,7 +880,13 @@ Strategy.prototype._authCodeFlowHandler = function authCodeFlowHandler(code, req
 
         log.info('received id_token: ', id_token);
 
-        return self._validateIdToken(id_token, null, access_token, req, (jwtClaimsStr, jwtClaims) => {
+        return self._validateResponse(id_token, null, access_token, req, (jwtClaimsStr, jwtClaims) => {
+          // for 'code id_token' flow, check iss/sub from the first id_token with those from this id_token
+          if (issFromPrevIdToken && issFromPrevIdToken !== jwtClaims.iss)
+            return self.fail('iss from the second id_token does not match iss from the first id_token');
```

Comment:
Instead of using first and second, you can use something like:
iss in the id_token from the token endpoint does not match the iss in the id_token from the authorization endpoint.